src/lib/core/media-trigger/media-trigger.ts

```diff
@@ -174,13 +172,14 @@ export class MediaTrigger {
    * Restore original, 'true' registry
    */
   private restoreRegistryMatches() {
-    const target = this.matchMedia.registry;
+    const target = new Map(this.matchMedia.registry);
```

Comment:
This change is no longer necessary. In fact, the below `target.clear()` makes cloning the registry rather unnecessary.

Either these changes can be reverted in full, or the clone + clear can be refactored. I would personally revert these changes.